src/View/Requirements_Backend.php

```diff
@@ -1421,6 +1440,32 @@ function () use ($fileList, $minify, $type) {
         return $combinedURL;
     }
 
+    /**
+     * Resolves relative paths in CSS files which are lost when combining them
+     *
+     * @param string $content
+     * @param string $filePath
+     * @return string New content with paths resolved
+     */
+    protected function resolveCSSReferences($content, $filePath)
+    {
+        $doResolving = Config::inst()->get(__CLASS__, 'resolve_relative_css_refs');
+        if (!$doResolving) {
+            return $content;
+        }
+        $fileUrl = Injector::inst()->get(ResourceURLGenerator::class)->urlForResource($filePath);
+        $fileUrlDir = dirname($fileUrl);
+        $content = preg_replace_callback('#(url\([\n\r\s\'"]*)([^\s\)\?\'"]+)#i', function ($match) use ($fileUrlDir) {
+            [ $_, $prefix, $relativePath ] = $match;
+            if ($relativePath[0] === '/' || false !== strpos($relativePath, '://')) {
+                return $prefix . $relativePath;
+            }
+            $full = FilesystemPath::canonicalize($fileUrlDir . '/' . $relativePath);
```

Comment:
```suggestion
            $full = FilesystemPath::canonicalize(FilesystemPath::join($fileUrlDir, $relativePath));
```
Silverstripe is pretty inconsistent at the moment about when it does or doesn't include a trailing slash. There's an open issue to resolve this, so a future major release may include a trailing slash in the result of `ResourceURLGenerator::urlForResource`.
Using `Path::join()` here now ensures that won't cause a regression later.